### DISCUSSION TITLE
Firing events does not update actor state in React

### DISCUSSION THREAD

**Discussion by Kashif7** [statelyai/xstate#2209] [Q&A] [answered]
https://codesandbox.io/s/vibrant-cookies-oru88

I am trying to spawn some actors and update their state using the send action creator returned by the "useActor" hook. But for some reason, the actor's state is not getting updated.  The parent machine works as expected.

**Comment by Kashif7** [comment, marked as answer]
Embarrassing mistake. Didn't use assign to update the context with the actors and that resulted in detached actors. Was trying to figure it out for a long time :'(.

**Reply by davidkpiano** [reply]
Sorry! Perhaps we can make dev-time analysis tools to help surface these issues more quickly.

**Reply by Kashif7** [reply]
That would of course help, but the documentation is quite comprehensive. That helps massively.
